File: nj2k/testData/newJ2k/methodCallExpression/specialBuiltinMembers.java
import java.util.*;

enum E {
    A, B, C
}

class A {
    void foo(List<String> list, Collection<Integer> collection, Map<Integer, Integer> map) {
        int a = "".length();
        String b = E.A.name();
        int c = E.A.ordinal();
        int d = list.size() + collection.size();
        int e = map.size();
        int f = map.keySet().size();
        int g = map.values().size();
        int h = map.entrySet().size();
        int i = map.entrySet().iterator().next().getKey() + 1
    }

    void bar(List<String> list, HashMap<String, Integer> map) {
        char c = "a".charAt(0);
        byte b = new Integer(10).byteValue();
        int i = new Double(10.1).intValue();
        float f = new Double(10.1).floatValue();
        long l = new Double(10.1).longValue();
        short s = new Double(10.1).shortValue();

        try {
            String removed = list.remove(10);
            Boolean isRemoved = list.remove("a");
        }
        catch(Exception e) {
            System.err.println(e.getMessage());
            throw new RuntimeException(e.getCause());
        }

        for (Map.Entry<String, Integer> entry : map.entrySet()) {
            String key = entry.getKey();
            Integer value = entry.getValue();
            entry.setValue(value + 1);
        }
    }

    void kt21504() {
        byte b = Byte.parseByte("1");
        short s = Short.parseShort("1");
        int i  = Integer.parseInt("1");
        long l = Long.parseLong("1");
        float f = Float.parseFloat("1");
        double d = Double.parseDouble("1");

        byte b2 = Byte.parseByte("1", 10);
        short s2 = Short.parseShort("1", 10);
        int i2  = Integer.parseInt("1", 10);
        long l2 = Long.parseLong("1", 10);
    }

    void kt7940() {
        byte b1 = Byte.MIN_VALUE;
        byte b2 = Byte.MAX_VALUE;
        short s1 = Short.MIN_VALUE;
        short s2 = Short.MAX_VALUE;
        int i1 = Integer.MIN_VALUE;
        int i2 = Integer.MAX_VALUE;
        long l1 = Long.MIN_VALUE;
        long l2 = Long.MAX_VALUE;
        float f1 = Float.MIN_VALUE;
        float f2 = Float.MAX_VALUE;
        float f3 = Float.POSITIVE_INFINITY;
        float f4 = Float.NEGATIVE_INFINITY;
        float f5 = Float.NaN;
        double d1 = Double.MIN_VALUE;
        double d2 = Double.MAX_VALUE;
        double d3 = Double.POSITIVE_INFINITY;
        double d4 = Double.NEGATIVE_INFINITY;
        double d5 = Double.NaN;
    }

    void kt35593() {
        Number number = 1;
        byte b = number.byteValue();
        double d = number.doubleValue();
        float f = number.floatValue();
        int i = number.intValue();
        long l = number.longValue();
        short s = number.shortValue();
    }
}

File: nj2k/testData/newJ2k/methodCallExpression/specialBuiltinMembers.kt
import java.util.HashMap

internal enum class E {
    A, B, C
}

internal class A {
    fun foo(list: List<String?>, collection: Collection<Int?>, map: Map<Int, Int?>) {
        val a = "".length
        val b = E.A.name
        val c = E.A.ordinal
        val d = list.size + collection.size
        val e = map.size
        val f = map.keys.size
        val g = map.values.size
        val h = map.entries.size
        val i = map.entries.iterator().next().key + 1
    }

    fun bar(list: MutableList<String?>, map: HashMap<String, Int>) {
        val c = "a"[0]
        val b = 10.toByte()
        val i = 10.1.toInt()
        val f = 10.1.toFloat()
        val l = 10.1.toLong()
        val s = 10.1.toInt().toShort()
        try {
            val removed = list.removeAt(10)
            val isRemoved = list.remove("a")
        } catch (e: Exception) {
            System.err.println(e.message)
            throw RuntimeException(e.cause)
        }
        for (entry in map.entries) {
            val key = entry.key
            val value = entry.value
            entry.setValue(value + 1)
        }
    }

    fun kt21504() {
        val b = "1".toByte()
        val s = "1".toShort()
        val i = "1".toInt()
        val l = "1".toLong()
        val f = "1".toFloat()
        val d = "1".toDouble()
        val b2 = "1".toByte(10)
        val s2 = "1".toShort(10)
        val i2 = "1".toInt(10)
        val l2 = "1".toLong(10)
    }

    fun kt7940() {
        val b1 = Byte.MIN_VALUE
        val b2 = Byte.MAX_VALUE
        val s1 = Short.MIN_VALUE
        val s2 = Short.MAX_VALUE
        val i1 = Int.MIN_VALUE
        val i2 = Int.MAX_VALUE
        val l1 = Long.MIN_VALUE
        val l2 = Long.MAX_VALUE
        val f1 = Float.MIN_VALUE
        val f2 = Float.MAX_VALUE
        val f3 = Float.POSITIVE_INFINITY
        val f4 = Float.NEGATIVE_INFINITY
        val f5 = Float.NaN
        val d1 = Double.MIN_VALUE
        val d2 = Double.MAX_VALUE
        val d3 = Double.POSITIVE_INFINITY
        val d4 = Double.NEGATIVE_INFINITY
        val d5 = Double.NaN
    }

    fun kt35593() {
        val number: Number = 1
        val b = number.toByte()
        val d = number.toDouble()
        val f = number.toFloat()
        val i = number.toInt()
        val l = number.toLong()
        val s = number.toShort()
    }
}